app/serializers/term_serializer.rb

```diff
@@ -5,9 +5,15 @@ class TermSerializer < ApplicationSerializer
   has_one :property
   has_many :vocabularies, serializer: PreviewSerializer
   has_one :organization, if: -> { params[:spine] || params[:with_organization] }, serializer: PreviewSerializer
-  attribute :max_mapping_weight do
-    params[:spine] ? object.max_mapping_weight : nil
+
+  attribute :max_mapping_weight, if: -> { params[:spine] } do
+    object.max_mapping_weight
+  end
+
+  attribute :current_mapping_weight, if: -> { params[:spine] } do
+    object.alignments.where.not(predicate_id: nil).joins(:predicate).sum("predicates.weight")
```

Comment:
The `NOT NULL` condition is needless here because the `INNER JOIN` does its job by returning matching records only (assuming that no predicate has `NULL` as its `id`).

Reply:
indeed, thank you - will update it at another branch based on this one